modules/programs/msmtp-accounts.nix

```diff
@@ -0,0 +1,40 @@
+{ pkgs }:
+{ config, lib, ... }:
+
+with lib;
+
+{
+  options.msmtp = {
+    enable = mkEnableOption "msmtp as sendmail";
+
+    command = {
+      send = mkOption {
+        readOnly = true;
+        type = types.str;
+        defaultText = "msmtp --account=<account name>";
+        description = ''
+          Command to send a message using this account via
+          <command>msmtp</command>.
+        '';
+      };
+
+      queue = mkOption {
+        readOnly = true;
```

Comment:
How do you plan to use queue vs send ? IMHO queue > send, aka if it can send the mail it will, else it will queue it which is what 99% people expect.

Reply:
To be honest I didn't really have any good plan with this, I mainly wanted to avoid the `offlineSendMethod` option 🙂

For now I'm inclined to entirely remove the command option and simply add a note in the module description to say that the msmtp account name will be the same as the account name in the `config.accounts.email.accounts` option. So users would have to decide themselves which command they would use and put the `--account` argument themselves. Similar to how the mbsync does it.

Then maybe can think of some nice consistent way to do this for all email account modules…